yroots/subdivision.py

```diff
@@ -272,7 +276,8 @@ def interval_approximate_1d(f,a,b,deg,inf_norm=None, return_bools=False):
     coeffs[deg]/=2
 
     if return_bools:
-        sign_change = not(np.all(values > 0) or np.all(values < 0))
+        # Check to see if the sign changes on the interval
+        sign_change = np.abs(np.sum(np.sign(values))) != 2*deg
```

Comment:
I was having some issues in your testing notebook here bc I was was confusing the degrees, but I guess 2*deg is correct bc this is using the more dense evaluation grid. However, when I was running the timing, this was slower than
```suggestion
     is_positive = values > 0
     sign_change = any(is_positive) and any(~is_positive)```